contrib/sddpg/core.c

```diff
@@ -169,12 +170,25 @@ static HB_ERRCODE pgsqlConnect( SQLDDCONNECTION * pConnection, PHB_ITEM pItem )
    PGconn *       pConn;
    ConnStatusType status;
    const char *   pszHost;
+   void **        hbConn;
+   PHB_ITEM       pSecond;
+   int existingConnection = 0;
 
-   pszHost = hb_arrayGetCPtr( pItem, 2 );
-   if( pszHost && ( strncmp( pszHost, "postgresql://", 13 ) == 0 || strchr( pszHost, '=' ) ) )
-      pConn = PQconnectdb( pszHost );
+   pSecond = hb_itemArrayGet( pItem, 2 );
+   if( HB_IS_POINTER( pSecond ) )
```

Comment:
Thank you for fixing all the stuff so far. There is also a bit that I'm not familiar with, adding/removing a reference of a GC collected pointer to an internal structure. Which means I'll have to pass on this one as well. You can keep the patch here though, in the hope someone can finish it up. It'd also be nice to squash it into a single commit (using `git rebase -i HEAD~5`).